pkg/stub/imagestreams.go

```diff
@@ -503,38 +520,34 @@ func (h *Handler) processImportStatus(is *imagev1.ImageStream, cfg *v1.Config) (
 		logrus.Debugf("no error/progress checks cause stream %s is skipped", is.Name)
 		// but if skipped, clear out any errors, since we do not care about errors for skipped
 		h.clearStreamFromImportError(is.Name, importError, cfg)
+		anyConditionUpdate = true
 	}
 
 	processing := util.Condition(cfg, v1.ImageChangesInProgress)
 
-	//logrus.Debugf("pending is %v any errors %v for %s", pending, anyErrors, is.Name)
 	logrus.Debugf("any errors %v for %s", anyErrors, is.Name)
 
 	// we check for processing == true here as well to avoid churn on relists
-	//if !pending {
-		if !anyErrors {
-			h.clearStreamFromImportError(is.Name, importError, cfg)
-		}
-		logrus.Printf("There are no more image imports in flight for imagestream %s", is.Name)
-		if processing.Status == corev1.ConditionTrue {
-			now := kapis.Now()
-			// remove this imagestream name, including the space separator, from processing
-			logrus.Debugf("current reason %s ", processing.Reason)
-			replaceOccurs := util.NameInReason(cfg, processing.Reason, is.Name)
-			if replaceOccurs {
-				processing.Reason = util.ClearNameInReason(cfg, processing.Reason, is.Name)
-				logrus.Debugf("processing reason now %s", processing.Reason)
-				if len(strings.TrimSpace(processing.Reason)) == 0 && processing.Status != corev1.ConditionFalse {
-					logrus.Println("The last in progress imagestream has completed (import status check)")
-					processing.Status = corev1.ConditionFalse
-					processing.Reason = ""
-				}
-				processing.LastTransitionTime = now
-				processing.LastUpdateTime = now
-				util.ConditionUpdate(cfg, processing)
-				anyConditionUpdate = true
+	logrus.Printf("There are no more image imports in flight for imagestream %s", is.Name)
+	if processing.Status == corev1.ConditionTrue {
+		now := kapis.Now()
+		// remove this imagestream name, including the space separator, from processing
+		logrus.Debugf("current reason %s ", processing.Reason)
+		replaceOccurs := util.NameInReason(cfg, processing.Reason, is.Name)
+		if replaceOccurs {
+			processing.Reason = util.ClearNameInReason(cfg, processing.Reason, is.Name)
+			logrus.Debugf("processing reason now %s", processing.Reason)
+			if len(strings.TrimSpace(processing.Reason)) == 0 && processing.Status != corev1.ConditionFalse {
+				logrus.Println("The last in progress imagestream has completed (import status check)")
+				processing.Status = corev1.ConditionFalse
+				processing.Reason = ""
 			}
+			processing.LastTransitionTime = now
```

Comment:
I think this belongs in the `if` block above or needs a separate conditional - not clear that we're guaranteed to transition status from `True` to `False` here.

Reply:
yep within the if block thanks

Reply:
update pushed @adambkaplan